### PR TITLE
improved start and end date validation for courses.

### LINKED ISSUE #1002: Need to Validate that Course End Date > Course Start Date
Currently, there is no problem saving a Course with a Start Date that occurs after the End Date.  We should fix this.

See screenshot for reference.

chumbawumba

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #1002

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit f721f17e0f495d9794101e2157693bfc1a40da8c

https://deploy-preview-2482--ilios-frontend.netlify.com

File: app/components/course-overview.js [hunk 1]
```diff
@@ -17,10 +17,16 @@ const Validations = buildValidations({
     }),
   ],
   startDate: [
-    validator('presence', true),
+    validator('date', {
+      dependentKeys: ['model.endDate'],
+      before: reads('model.endDate'),
```
Comment (jrjohnson) [inline app/components/course-overview.js]: I'm thinking this should be onOrBefore to allow for courses to start / end on the same day.

File: app/components/course-overview.js [hunk 2]
```diff
@@ -17,10 +17,16 @@ const Validations = buildValidations({
...
+    }),
   ],
   endDate: [
-    validator('presence', true),
+    validator('date', {
+      dependentKeys: ['model.startDate'],
+      after: reads('model.startDate'),
```
Comment (jrjohnson) [inline app/components/course-overview.js]: I'm thinking this should be onOrAfter to allow for courses to start / end on the same day.

File: app/components/course-overview.js (hunk 1 above)
Comment (stopfstedt) [inline app/components/course-overview.js]: handled.

File: app/components/course-overview.js (hunk 2 above)
Comment (stopfstedt) [inline app/components/course-overview.js]: handled.

**Pull request merged**